validate/validate.go

```diff
@@ -309,31 +309,40 @@ func (v *Validator) CheckMounts() (msgs []string) {
 func (v *Validator) CheckLinux() (msgs []string) {
 	logrus.Debugf("check linux")
 
-	utsExists := false
-	ipcExists := false
-	mountExists := false
-	netExists := false
-	userExists := false
+	var typeList = map[rspec.NamespaceType]struct {
+		num      int
+		newExist bool
+	}{
+		rspec.PIDNamespace:     {0, false},
+		rspec.NetworkNamespace: {0, false},
+		rspec.MountNamespace:   {0, false},
+		rspec.IPCNamespace:     {0, false},
+		rspec.UTSNamespace:     {0, false},
+		rspec.UserNamespace:    {0, false},
+		rspec.CgroupNamespace:  {0, false},
+	}
 
 	for index := 0; index < len(v.spec.Linux.Namespaces); index++ {
 		if !namespaceValid(v.spec.Linux.Namespaces[index]) {
 			msgs = append(msgs, fmt.Sprintf("namespace %v is invalid.", v.spec.Linux.Namespaces[index]))
-		} else if len(v.spec.Linux.Namespaces[index].Path) == 0 {
-			if v.spec.Linux.Namespaces[index].Type == rspec.UTSNamespace {
-				utsExists = true
-			} else if v.spec.Linux.Namespaces[index].Type == rspec.IPCNamespace {
-				ipcExists = true
-			} else if v.spec.Linux.Namespaces[index].Type == rspec.NetworkNamespace {
-				netExists = true
-			} else if v.spec.Linux.Namespaces[index].Type == rspec.MountNamespace {
-				mountExists = true
-			} else if v.spec.Linux.Namespaces[index].Type == rspec.UserNamespace {
-				userExists = true
-			}
 		}
+
+		nsType := v.spec.Linux.Namespaces[index].Type
+		nsPath := v.spec.Linux.Namespaces[index].Path
```

Comment:
nit: I'd rather have `ns := v.spec.Linux.Namespaces[index]`.  Only one variable (vs. the two here), and you only need an extra character when you use it (e.g. `ns.Type` vs your `nsType`).